[CI] TransportFetchPhaseCoordinationActionTests testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming failing

**Build Scans:**
- [elasticsearch-intake #39557 / part1](https://gradle-enterprise.elastic.co/s/jqricwqsnmd6e)
- [elasticsearch-intake #39549 / part1](https://gradle-enterprise.elastic.co/s/umsotnxmwtomg)
- [elasticsearch-intake #39519 / part1](https://gradle-enterprise.elastic.co/s/oafjz44v5uuza)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.search.fetch.chunk.TransportFetchPhaseCoordinationActionTests.testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming" -Dtests.seed=60884BD4B69F6489 -Dtests.locale=en-PW -Dtests.timezone=Asia/Anadyr -Druntime.java=26
```

**Applicable branches:**
9.4

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.search.fetch.chunk.TransportFetchPhaseCoordinationActionTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <0L>
     but: was <45L>
```

**Issue Reasons:**
- [9.4] 1 consecutive failures in test testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming
- [9.4] 1 consecutive failures in step part1
- [9.4] 1 consecutive failures in pipeline elasticsearch-intake
- [9.4] 3 failures in test testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming (4.8% fail rate in 63 executions)
- [9.4] 3 failures in step part1 (18.8% fail rate in 16 executions)
- [9.4] 3 failures in pipeline elasticsearch-intake (18.8% fail rate in 16 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -322,3 +322,4 @@ tests:
 #    issue: "https://github.com/elastic/elasticsearch/..."
 #  - class: "org.elasticsearch.xpack.esql.**"
 #    method: "test {union_types.MultiIndexIpStringStatsInline *}"
+#    issue: "https://github.com/elastic/elasticsearch/..."

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -285,18 +285,6 @@ tests:
 - class: org.elasticsearch.search.KnnSearchIT
   method: testKnnSearchWithScroll
   issue: https://github.com/elastic/elasticsearch/issues/148590
-- class: org.elasticsearch.search.fetch.chunk.TransportFetchPhaseCoordinationActionTests
-  method: testActionType
-  issue: https://github.com/elastic/elasticsearch/issues/148911
-- class: org.elasticsearch.search.fetch.chunk.TransportFetchPhaseCoordinationActionTests
-  method: testDoExecuteReleasesRegistrationOnLastChunkDeserializationFailure
-  issue: https://github.com/elastic/elasticsearch/issues/148912
-- class: org.elasticsearch.search.fetch.chunk.TransportFetchPhaseCoordinationActionTests
-  method: testDoExecuteReleasesRegistrationWhenDataNodeFailsAfterChunkStreaming
-  issue: https://github.com/elastic/elasticsearch/issues/148913
-- class: org.elasticsearch.search.rescore.QueryRescoreModeTests
-  method: testQueryRescoreMode
-  issue: https://github.com/elastic/elasticsearch/issues/148923
 
 # Examples:
 #

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -322,4 +322,3 @@ tests:
 #    issue: "https://github.com/elastic/elasticsearch/..."
 #  - class: "org.elasticsearch.xpack.esql.**"
 #    method: "test {union_types.MultiIndexIpStringStatsInline *}"
-#    issue: "https://github.com/elastic/elasticsearch/..."

--- a/server/src/test/java/org/elasticsearch/search/fetch/chunk/TransportFetchPhaseCoordinationActionTests.java
+++ b/server/src/test/java/org/elasticsearch/search/fetch/chunk/TransportFetchPhaseCoordinationActionTests.java
@@ -18,6 +18,7 @@ import org.elasticsearch.action.support.ActionFilters;
 import org.elasticsearch.action.support.IndicesOptions;
 import org.elasticsearch.action.support.PlainActionFuture;
 import org.elasticsearch.cluster.node.VersionInformation;
+import org.elasticsearch.common.breaker.CircuitBreaker;
 import org.elasticsearch.common.bytes.BytesArray;
 import org.elasticsearch.common.bytes.BytesReference;
 import org.elasticsearch.common.bytes.ReleasableBytesReference;
@@ -75,6 +76,7 @@ public class TransportFetchPhaseCoordinationActionTests extends ESTestCase {
     private ActiveFetchPhaseTasks activeFetchPhaseTasks;
     private NamedWriteableRegistry namedWriteableRegistry;
     private TransportFetchPhaseCoordinationAction action;
+    private CircuitBreakerService breakerService;
 
     @Before
     public void setUp() throws Exception {
@@ -92,7 +94,7 @@ public class TransportFetchPhaseCoordinationActionTests extends ESTestCase {
         activeFetchPhaseTasks = new ActiveFetchPhaseTasks();
         namedWriteableRegistry = new NamedWriteableRegistry(Collections.emptyList());
 
-        CircuitBreakerService breakerService = newLimitedBreakerService(ByteSizeValue.ofMb(64));
+        breakerService = newLimitedBreakerService(ByteSizeValue.ofMb(64));
         action = new TransportFetchPhaseCoordinationAction(
             transportService,
             new ActionFilters(Set.of()),
@@ -381,9 +383,20 @@ public class TransportFetchPhaseCoordinationActionTests extends ESTestCase {
         action.doExecute(createTask(taskId), request, future);
         expectThrows(Exception.class, () -> future.actionGet(10, TimeUnit.SECONDS));
 
-        assertBusy(() -> {
-            expectThrows(ResourceNotFoundException.class, () -> activeFetchPhaseTasks.acquireResponseStream(taskId, TEST_SHARD_ID));
-        });
+        // doExecute adds bytes to the REQUEST breaker before SearchHit.readFrom() throws on the
+        // invalid payload. closeInternal (triggered when the stream's refCount reaches zero after
+        // cleanup) releases those bytes, so zero bytes is the reliable signal that the stream is
+        // fully cleaned up. Polling acquireResponseStream instead would tryIncRef on every failed
+        // attempt and leak refs, preventing closeInternal from ever firing.
+        assertBusy(
+            () -> assertThat(
+                "breaker bytes must be zero once stream closeInternal has completed",
+                breakerService.getBreaker(CircuitBreaker.REQUEST).getUsed(),
+                equalTo(0L)
+            )
+        );
+        // Confirm the task was also deregistered (single call, no retry loop, no ref leak).
+        expectThrows(ResourceNotFoundException.class, () -> activeFetchPhaseTasks.acquireResponseStream(taskId, TEST_SHARD_ID));
     }
 
     public void testDoExecutePreservesContextIdInFinalResult() throws Exception {
@@ -470,9 +483,21 @@ public class TransportFetchPhaseCoordinationActionTests extends ESTestCase {
         Exception failure = expectThrows(Exception.class, () -> future.actionGet(10, TimeUnit.SECONDS));
         assertThat(failure.getMessage(), equalTo("simulated data node failure during chunk streaming"));
 
+        // Wait until closeInternal fires so tearDown's transportService.close() cannot race with
+        // processChunk.finally still holding the stream ref. The circuit breaker is released in
+        // closeInternal, so zero bytes means the stream has been fully cleaned up (cleanup ran
+        // AND processChunk.finally ran AND closeInternal was triggered). Using the breaker check
+        // instead of polling acquireResponseStream, which would tryIncRef on every failed poll and
+        // leak refs, preventing closeInternal from ever being triggered.
         assertBusy(
-            () -> expectThrows(ResourceNotFoundException.class, () -> activeFetchPhaseTasks.acquireResponseStream(taskId, TEST_SHARD_ID))
+            () -> assertThat(
+                "breaker bytes must be zero once stream closeInternal has completed",
+                breakerService.getBreaker(CircuitBreaker.REQUEST).getUsed(),
+                equalTo(0L)
+            )
         );
+        // Confirm the task was also deregistered (single call, no retry loop, no ref leak).
+        expectThrows(ResourceNotFoundException.class, () -> activeFetchPhaseTasks.acquireResponseStream(taskId, TEST_SHARD_ID));
     }
 
     private ShardFetchSearchRequest createShardFetchSearchRequest() {

--- a/server/src/test/java/org/elasticsearch/search/fetch/chunk/TransportFetchPhaseResponseChunkActionTests.java
+++ b/server/src/test/java/org/elasticsearch/search/fetch/chunk/TransportFetchPhaseResponseChunkActionTests.java
@@ -256,7 +256,10 @@ public class TransportFetchPhaseResponseChunkActionTests extends ESTestCase {
             originalHit.decRef();
         }
 
-        assertThat("breaker bytes should be released when stream is closed", breaker.getUsed(), equalTo(0L));
+        // processChunk.finally (responseStream.decRef) may still be pending on a transport thread
+        // when the test's own stream.decRef runs. assertBusy waits until closeInternal fires
+        // (observable via breaker bytes = 0) before returning, preventing tearDown from racing it.
+        assertBusy(() -> assertThat("breaker bytes should be released when stream is closed", breaker.getUsed(), equalTo(0L)));
     }
 
     private SearchHit createHit(int id) {

